src/PrivateDns/PrivateDns/Zones/SetAzurePrivateDnsZone.cs

```diff
@@ -0,0 +1,117 @@
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.PrivateDns.Zones
+{
+    using System.Collections;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.PrivateDns.Models;
+    using Microsoft.Azure.Commands.PrivateDns.Utilities;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using ProjectResources = Microsoft.Azure.Commands.PrivateDns.Properties.Resources;
+
+    /// <summary>
+    /// Updates an existing zone.
+    /// </summary>
+    [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "PrivateDnsZone", SupportsShouldProcess = true, DefaultParameterSetName = FieldsParameterSetName), OutputType(typeof(PSPrivateDnsZone))]
+    public class SetAzurePrivateDnsZone : PrivateDnsBaseCmdlet
+    {
+        private const string FieldsParameterSetName = "Fields";
+        private const string ObjectParameterSetName = "Object";
+        private const string ResourceParameterSetName = "ResourceId";
+
+        [Parameter(Mandatory = true, HelpMessage = "The resource group in which the zone exists.", ParameterSetName = FieldsParameterSetName)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The full name of the zone (without a terminating dot).", ParameterSetName = FieldsParameterSetName)]
+        [ResourceNameCompleter("Microsoft.Network/privateDnsZones", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(ParameterSetName = ResourceParameterSetName, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Private DNS Zone ResourceID.")]
+        [ResourceIdCompleter("Microsoft.Network/privateDnsZones")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, HelpMessage = "The zone object to set.", ParameterSetName = ObjectParameterSetName)]
+        [ValidateNotNullOrEmpty]
+        public PSPrivateDnsZone PrivateZone { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "A hash table which represents resource tags.", ParameterSetName = FieldsParameterSetName)]
+        [Parameter(Mandatory = false, HelpMessage = "A hash table which represents resource tags.", ParameterSetName = ResourceParameterSetName)]
+        public Hashtable Tag { get; set; }
```

Comment:
Please add the Tag parameter to the object parameter set as well - if the user wants to override the value in the object (aka, just use the object for identity parameters) they can specify -Tag, otherwise we will set using the Tag value in the object.  This will enable scenarios such as setting a tag for all zones in a resourcegroup without having to do unnecessary object manipulation that complicates the piping scenario:
```
Get-AzPrivateDnsZone -ResourceGroupName maddie | Set-AzPrivateDnsZone -Tag @{"a" = "b"}
```
This is how all new Set cmdlets look in our repo, so we want users coming from other modules to have a logical transition over to your cmdlets.

Reply:
changed it.